### PR TITLE
Fractional Knapsack in Java

### PR DESCRIPTION
**Pull request by Coder-Srinivas**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Fractional Knapsack in Java's Dynamic Programming Section

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue : #2347

### CODE REVIEW

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+// Output
+
+// 150
+
+// Explanation
+
+// Take the item with weight 5 and value 100, we are left with capacity 3
+// Take half of the item with weight 6, so it's value would become 50.
+// Thereby getting a maximum profit of 150
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Few changes: 
- Use multi-line comments 
- Shift sample I/O to the end of the code
- Add complexity at the end of the Code

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+import java.util.*;
+
+
+// Class Fractional Knapsack
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        // Inputting the values
+        capacity = input.nextInt();
+
+        input.close();
+
+        // Printing the max profit
+        System.out.println(getMaxProfit(weights, values, capacity));
+    }
+    /**End Main Function**/  
+}
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Add a new line at the EOF

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+        // Inputting the values
+        capacity = input.nextInt();
+
+        input.close();
+
+        // Printing the max profit
+        System.out.println(getMaxProfit(weights, values, capacity));
+    }
+    /**End Main Function**/
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+        // Initializing the variable capacity
+        int capacity = 0;
+
+        // Inputting the weights
+        for(int i=0; i<items; i++){
+            weights[i] = input.nextInt();
+        }
+
+        // Inputting the values
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        for(int i=0; i<items; i++){
+            values[i] = input.nextInt();
+        }
+
+        // Inputting the values
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+                //Breaking from loop as we reached our peak capacity
+                break;
+            }
+        }
+
+        return ans;
+    }
+
+    // Main method
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    public static void main(String[] args) {
+        
+        Scanner input = new Scanner(System.in);
+
+        // Taking the number of items from the user
+        int items = input.nextInt();
+
+        // Initializing an empty weights and values array
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+        return ans;
+    }
+
+    // Main method
+    public static void main(String[] args) {
+        
+        Scanner input = new Scanner(System.in);
+
+        // Taking the number of items from the user
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

File: Java/Greedy_Algorithms/FractionalKnapsack.java [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+        int items = input.nextInt();
+
+        // Initializing an empty weights and values array
+        int weights[] = new int[items];
+        int values[] = new int[items];
+
+        // Initializing the variable capacity
```
Comment (HarshCasper) [inline Java/Greedy_Algorithms/FractionalKnapsack.java]: Redundant Comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Coder-Srinivas** [comment]
@HarshCasper Made necessary changes. Please review it and let me know if anything is wrong ☺.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**